UnitySDK/Assets/ML-Agents/Scripts/DemonstrationStore.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.IO;
+using UnityEngine;
+
+
+namespace MLAgents
+{
+    /// <summary>
+    /// Demonstration meta-data.
+    /// </summary>
+    [System.Serializable]
+    public struct DemonstrationMetaData
+    {
+        public int numberExperiences;
+        public int numberEpisodes;
+        public int API;
+    }
+    
+    public class DemonstrationStore
+    {
+        private string filePath;
+        private DemonstrationMetaData metaData;
+        
+        
+        /// <summary>
+        /// Checks for the existence of the Demonstrations directory
+        /// and creates it if it does not exist.
+        /// </summary>
+        public void CreateDirectory()
+        {
+            if (!Directory.Exists("Assets/Demonstrations"))
+            {
+                Directory.CreateDirectory("Assets/Demonstrations");
+            }
+        }
+        
+        /// <summary>
+        /// Creates demonstration file, and writes brainParameters as json to file.
+        /// </summary>
+        public void CreateDemonstrationFile(string demonstrationName)
+        {
+            // Creates demonstration file.
+            var literalName = demonstrationName;
+            filePath = "Assets/Demonstrations/" + literalName + ".demo";
+            var uniqueNameCounter = 0;
+            while (File.Exists(filePath))
+            {
+                literalName = demonstrationName + "_" + uniqueNameCounter;
+                filePath = "Assets/Demonstrations/" + literalName + ".demo";
+                uniqueNameCounter++;
+            }
+            
+            metaData = new DemonstrationMetaData
+            {
+                numberEpisodes = 0,
+                numberExperiences = 0,
+                API = 1
+            };
+        }
+
+        /// <summary>
+        /// Writes brain parameters as json to file.
```

Comment:
As much as possible, method-level comments should not expose implementation details (e.g. saving to file, JSON format)- it should be more of the what than how. This comment applies to a few method docs elsewhere.